narwhal/primary/src/state_handler.rs

```diff
@@ -45,11 +63,29 @@ pub struct StateHandler {
 // Internal state for randomness DKG and generation.
 // TODO: Write a brief protocol description.
 struct RandomnessState {
+    // A channel to send system messages to the proposer.
+    tx_system_messages: Sender<SystemMessage>,
+
+    // State for DKG.
     party: dkg::Party<PkG, EncG>,
     processed_messages: Vec<dkg::ProcessedMessage<PkG, EncG>>,
     used_messages: Option<dkg::UsedProcessedMessages<PkG, EncG>>,
     confirmations: Vec<dkg::Confirmation<EncG>>,
     dkg_output: Option<dkg::Output<PkG, EncG>>,
+    vss_key_output: Arc<OnceLock<PublicVssKey>>,
+
+    // State for randomness generation.
+    authority_id: AuthorityIdentifier,
+    leader_schedule: LeaderSchedule,
+    network: anemo::Network,
+    last_randomness_round_sent: Option<RandomnessRound>,
+    randomness_round: RandomnessRound,
+    last_narwhal_round_sent: Round,
+    narhwal_round: Round,
+    cached_sigs: Option<(RandomnessRound, Vec<RandomnessPartialSignature>)>,
```

Comment:
Add a comment on when and why the sigs are cached?

Reply:
done